src/main/java/spoon/support/sniper/internal/AbstractSourceFragmentContextCollection.java

```diff
@@ -68,6 +71,17 @@ public boolean knowsHowToPrint(PrinterEvent event) {
 		throw new SpoonException("Unexpected PrintEvent: " + event.getClass());
 	}
 
+	private boolean childFragmentHasSpecifiedRoleInParent(CtRole roleInParent) {
+		Optional<SourceFragment> optionSourceFragment = childFragments.stream()
+				.filter(fragment -> fragment instanceof ElementSourceFragment)
+				.findFirst();
+		if (optionSourceFragment.isPresent()) {
+			ElementSourceFragment elementSourceFragment = (ElementSourceFragment) optionSourceFragment.get();
+			return  elementSourceFragment.getRoleInParent() == roleInParent;
+		}
+		return false;
+	}
```

Comment:
> I assumed that if the first element has the role CtRole.TRY_RESOURCE, then the rest elements will also have the same role and I guess it's safe to do so because you can only used objects inside it which implement

That's probably safe, yes. The thingy that builds the `CollectionSourceFragment` (which is later fed into the collection context) groups the children by role. I don't have a great understanding of how often this check will run, but I don't think a solitary `;` is printed very often. In the entire sniper printer unit test suite, it only happens 4 times, 2 of which are in your tests. So probably this is fine from a performance perspective, but it bears keeping in mind.

Reading the method title, I'm still thinking "which child fragment?". And the answer to that is of course _any_ child fragment. I would therefore suggest a renaming like below, and also further idiomatic use of the stream.

```suggestion
	private boolean anyChildFragmentHasRole(CtRole role) {
		return childFragments.stream()
				.filter(ElementSourceFragment.class::isInstance)
				.map(ElementSourceFragment.class::cast)
				.map(ElementSourceFragment::getRoleInParent)
				.map(role::equals)
				.findAny()
				.orElse(false);
	}
```

I'm using `findAny()` here to emphasize that there's nothing special about the first element, we just assume that if any element has a given role, the others do too.